src/System.Management.Automation/engine/regex.cs

```diff
@@ -278,6 +278,70 @@ public static bool ContainsWildcardCharacters(string pattern)
             return result;
         }
 
+        /// FIXME
+        /// For invalid wildcard pattern, it currently only check if there is
+        /// unclosed bracket. Need to find a better solution and rewrite this.
+        ///
+        /// <summary>
+        /// Checks to see if the given string might contains wildcard pattern.
+        /// </summary>
+        /// <param name="pattern">
+        /// String which needs to be checked
+        /// </param>
+        /// <returns>
+        /// true if the string does not contain invalid wildcard pattern,
+        /// false otherwise.
+        /// </returns>
+        /// <remarks>
+        /// Currently string contains { '*', '?' } or both { '[', ']' } is
+        /// considered to be properbly a wildcard pattern and
+        /// '`' is the escape character.
+        /// </remarks>
+        internal static bool ContainsValidWildcardPattern(string pattern)
+        {
+            if (string.IsNullOrEmpty(pattern))
+            {
+                return false;
+            }
+
+            bool hasWildcard = false;
+            bool openBracket = false;
+
+            for (int index = 0; index < pattern.Length; ++index)
+            {
+                switch (pattern[index])
+                {
+                    case '*':
+                    case '?':
+                        hasWildcard = true;
+                        break;
+
+                    case '[':
+                        hasWildcard = true;
```

Comment:
`ContainsValidWildcardPattern("du[[ck]")` will return `true`.

We probably need something like:

```csharp
case '[':
    if (openBracket)
    {
        return false;
    }
```

Reply:
`du[[ck]` should be a valid wildcard pattern.